### ISSUE TITLE
Ideas for more tutorials

### ISSUE THREAD

**Issue by helfer** [apollographql/hermes#2]
This is the thread for your ideas of other things I (or someone else) should make a (mini-) tutorial for. Please don't hold back!

**Comment by AkshayPathak** [comment]
I would like to know how to connect a SQL database to a GraphQL server using Sequalize or Objection.js

**Comment by AkshayPathak** [comment]
Also could you please start the tutorials with the server?

**Comment by AnthoniG** [comment]
I know these things are more biased towards the full stack. However I would also like to see more tutorials on different servers because not all of us can upload to JS enabled servers. Mainly PHP server (using their official PHP library of course) + other servers.

**Comment by helfer** [comment]
@AkshayPathak Good idea, I'll try to include that in the SQL part of the tutorial, or make a separate tutorial about it.

@AnthoniG Yeah, I think that could definitely be worth it. I was thinking about making one for Ruby and Graphene-Django, but if there's enough demand for a PHP version, I'll do that as well 👍

**Comment by tetiogit** [comment]
+1 for SQL. One of the parts lacking in all tutorials I've seen around is connection to a real sql database.

**Comment by pselden** [comment]
I'd like to see how GraphQL/Apollo can work in tandem with some other state management system.  I've been struggling internally about using apollo and react-apollo (and maybe even graphql) because in my ideal world GraphQL would be some implementation detail that is just used to load the data into a store and not be so tied at the component level -- but then you lose some/much of the magic that makes it so cool.

So maybe a lesson how how the two can work in tandem? I'm thinking of using it with MobX.

**Comment by ALawliet** [comment]
Unit testing, integration testing, continuous integration for a React + (Apollo) GraphQL app. 👍

**Comment by kfern** [comment]
+1 + TDD

**Comment by helfer** [comment]
@ALawliet @kfern what specifically would you want to learn from a TDD / CI tutorial?

**Comment by kfern** [comment]
Hi :-)

Maybe like https://github.com/dwyl/learn-tdd

**Comment by helfer** [comment]
@kfern Are there some GraphQL-specific things that you'd like to learn that don't work with the approach taught in the tutorial you linked to?

**Comment by kfern** [comment]
@helfer. I don't know. Do you think a generic tdd Javascript tutorial show what I need test and how with GraphQL?

**Comment by helfer** [comment]
@kfern yeah, I think in general the TDD concepts should be directly applicable to GraphQL. What I think we could provide is some guidance for how/when to test resolvers, and how to easily test components wrapped with `graphql` on the client.

**Comment by AnthoniG** [comment]
How about a tutorial that shows how to use GraphQL with MobX for example? It seems that Apollo uses Redux behind the scenes and expects you to hook up your store to that. However what if you don't want / can't use Redux ?

**Comment by helfer** [comment]
@AnthoniG We're working on a pluggable store API that will allow you to use Redux, Mobx or any other store behind the scenes. Once that's ready (in about 1-3 months), it would be great if someone wrote a tutorial for it! 🙂

**Comment by kfern** [comment]
@helfer. 👍

**Comment by booboothefool** [comment]
> What I think we could provide is some guidance for how/when to test resolvers, and how to easily test components wrapped with graphql on the client

@helfer That sounds amazing. 👍 Use Jest for the unit tests! Also, I'm kind of curious what tools you recommend for the integration tests and continuous integration. I'm looking at CircleCI/Travis.

Another idea: When/how to use https://github.com/facebook/dataloader with Apollo GraphQL.

**Comment by helfer** [comment]
@booboothefool we use both circle CI and Travis on different repos. I'd say they're both fine choices. I've never actually looked in detail at all the features they each provide.

**Comment by kfern** [comment]
Another Apollo tutorial: https://github.com/swarthout/feathers-apollo/blob/master/docs/explanation-blog.markdown
